src/packages/src/xroad/common/secret-store-local/usr/share/xroad/scripts/restore_openbao_db.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+abort() { local rc=$?; echo -e "FATAL: $*" >&2; exit $rc; }
+
+while getopts "F" opt ; do
+  case ${opt} in
+    F)
+      FORCE_RESTORE=true
+      ;;
+    \?)
+      echo "Invalid option $OPTARG -- did you use the correct wrapper script?"
+      exit 2
+      ;;
+  esac
+done
+
+shift $(($OPTIND - 1))
+
+dump_file="$1"
+
+db_connection_url=$(awk '/storage "postgresql"/,/\}/' /etc/openbao/openbao.hcl | grep 'connection_url' | cut -d'"' -f2)
+pat='^postgres:\/\/([^:]+):([^@]+)@([^:]+):([0-9]+)\/([^?]+)(\?(.*))?$'
+if [[ $db_connection_url =~ $pat ]]; then
+  db_conn_user="${BASH_REMATCH[1]}"
+  db_user=${db_conn_user%%@*}
+  db_password="${BASH_REMATCH[2]}"
+  db_addr="${BASH_REMATCH[3]}"
+  db_port="${BASH_REMATCH[4]}"
+  db_database="${BASH_REMATCH[5]}"
+  db_options="${BASH_REMATCH[7]}" # full query string after '?', if present
+  schema_pat='(^|&)search_path=([^&]*)'
+  if [[ $db_options =~ $schema_pat ]]; then
+    db_schema="${BASH_REMATCH[2]}"
+  else
+    db_schema="public"
+  fi
+  pg_options="-c client-min-messages=warning -c search_path=$db_schema"
+else
+  echo "Unable to determine OpenBao PostgreSQL connection URL in /etc/openbao/openbao.hcl"
+  exit 1
+fi
+
+# Reading custom libpq ENV variables
+if [ -f /etc/xroad/db_libpq.env ]; then
+  source /etc/xroad/db_libpq.env
+fi
+
+if [[ ! -z $PGOPTIONS_EXTRA ]]; then
+  PGOPTIONS_EXTRA=" ${PGOPTIONS_EXTRA}"
```

Comment:
PG env variables seem to be serverconf specific (for example messagelog does not use em). We might encounter some issues if we try reusing same vars.

Reply:
Same case as with db_libpq.env - PGOPTIONS_EXTRA are taken into account by all the databases during their setup - setup & restore processes should consistent.